src/ui/public/notify/__tests__/notifier.js

```diff
@@ -185,6 +197,60 @@ describe('Notifier', function () {
     });
   });
 
+  describe('#custom', function () {
+    let customNotif;
+
+    beforeEach(() => {
+      customNotif = notifier.custom(customText, customParams);
+    });
+
+    afterEach(() => {
+      customNotif.clear();
+    });
+
+    it('has a custom function to make notifications', function () {
+      expect(notifier.custom).to.be.defined;
+    });
+
+    it('properly merges options', function () {
+      expect(customNotif).to.have.property('title', customParams.title);
+      expect(customNotif).to.have.property('lifetime', customParams.lifetime);
+
+      expect(customParams.title).to.be.a('string');
+      expect(customParams.lifetime).to.be.a('number');
+    });
+
+    it('sets the content', function () {
+      expect(customNotif).to.have.property('content', `${params.location}: ${customText}`);
+      expect(customNotif.content).to.be.a('string');
+    });
+
+    it('uses custom actions', function () {
+      expect(customNotif).to.have.property('customActions');
+      expect(customNotif.customActions).to.have.length(customParams.actions.length);
+    });
+
+    it('gives a default action if none are provided', function () {
+      // destroy the default custom notification, avoid duplicate handling
+      customNotif.clear();
+
+      const noActionParams = _.defaults({ actions: [] }, customParams);
```

Comment:
If you want, you can use ES2015 syntax instead of lodash here:

```javascript
const noActionParams = Object.assign({}, customParams, { actions: [] });
```